### PR TITLE
🌟 DRAFT: Feature/db adapter

### PR DESCRIPTION
**Pull request by joao-victor-campos**

### CODE REVIEW

File: last_gas/adapters/db_adapters.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+from last_gas.domain.ports import DBLoader
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import sessionmaker
+from typing import Any, Dict
+
+Base = declarative_base()
```
Comment (victor-lucio) [inline last_gas/adapters/db_adapters.py:6]: not being used in the code

File: last_gas/adapters/db_adapters.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        self.session = Session()
+
+    def get(self, orm_class, id: int) -> Dict[str, Any]:
+        return {
+            key: val
+            for key, val in self.session.query(orm_class).get(id).__dict__.items()
+            if not key.startswith("_")
+        }
+
+    def get_all(self, orm_class, id: int) -> Dict[str, Any]:
```
Comment (victor-lucio) [inline last_gas/adapters/db_adapters.py]: we don't need id here, right?

File: last_gas/adapters/db_adapters.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        self.session.commit()
+
+    def delete(
+        self,
+        table_class: object,
+        primary_key_name: str,
+        id: int,
+    ) -> None:
+        self.session.query(table_class).filter(
+            getattr(table_class, primary_key_name) == id
```
Comment (victor-lucio) [inline last_gas/adapters/db_adapters.py:52]: not sure, but I think we can use the pk name directly to the filter, you can leave it like this, just a heads up.

File: last_gas/domain/commands/register_schedules.py [hunk 4]
```diff
@@ -1,16 +1,53 @@
...
+            self.schedule_type,
+        )
+        config_manager = EnvVarConfigLoader()
+        configs = config_manager.load_configs()
+        user = configs["USER"]
+        password = configs["PASSWORD"]
+
+        creator = EngineCreator(
+            database_flavour="postgresql",
+            host="localhost",
```
Comment (victor-lucio) [inline last_gas/domain/commands/register_schedules.py]: we should use an ENV var for that

File: Dockerfile [hunk 5]
```diff
@@ -3,6 +3,9 @@ FROM python:3.10.4
 # Install make
 RUN apt-get install -y make;
 
+# Install make
+RUN apt-get install -y make;
+
```
Comment (victor-lucio) [inline Dockerfile]: installing make twice.. btw, why do we need that in dockerfile?

File: last_gas/adapters/engine_creator.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
+from sqlalchemy import create_engine
+from sqlalchemy.engine import Engine
+
+
+class EngineCreator:
+    """Create a Engine for database connection."""
+
+    def __init__(
+        self,
+        database_flavour: str,
```
Comment (victor-lucio) [inline last_gas/adapters/engine_creator.py]: 🇬🇧

Comment (victor-lucio) [reply last_gas/adapters/engine_creator.py]: 😂

File: last_gas/domain/schedulers/orm.py [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+from sqlalchemy.orm import declarative_base, relationship
+
+Base = declarative_base()
+
+
+class SchedulesTypes(Base):
+    __tablename__ = "schedules_types"
+
+    schedule_type_id = Column("schedule_type_id", Integer, primary_key=True)
+    schedule_type = Column("schedule_type", String)
```
Comment (victor-lucio) [inline last_gas/domain/schedulers/orm.py]: should we have constraints for that? I mean, only defined values

File: last_gas/domain/commands/register_schedules.py [hunk 8]
```diff
@@ -1,16 +1,53 @@
 import discord
 from discord.ext import commands
 from discord import app_commands
+import uuid
+from datetime import datetime
+
+from last_gas.adapters.engine_creator import EngineCreator
+from last_gas.adapters.db_adapters import PostgresLoader
+from last_gas.adapters import EnvVarConfigLoader
+from last_gas.domain.schedulers import Schedules, SchedulesTypes
```
Comment (victor-lucio) [inline last_gas/domain/commands/register_schedules.py]: I think it's not gonna fins those classes, it's not in schedulers init

File: last_gas/domain/commands/register_schedules.py (hunk 4 above)
Comment (joao-victor-campos) [reply last_gas/domain/commands/register_schedules.py]: Noted! Next commit will be removing this class and suing EnvLoader instead

File: last_gas/adapters/db_adapters.py (hunk 2 above)
Comment (joao-victor-campos) [reply last_gas/adapters/db_adapters.py]: Yep

File: last_gas/domain/schedulers/orm.py (hunk 7 above)
Comment (joao-victor-campos) [reply last_gas/domain/schedulers/orm.py]: Wouldn't it be better to scale if we were to add new schedule_types to it?

File: Dockerfile (hunk 5 above)
Comment (joao-victor-campos) [reply Dockerfile]: would'nt make be needed for 
```
WORKDIR last_gas/
COPY . .
RUN make requirements
```

**Pull request merged**